### PR TITLE
BUG: infer_dtype with ArrowDtype

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/extension/test_arrow.py [hunk 1]
```diff
@@ -2875,3 +2875,15 @@ def test_duration_overflow_from_ndarray_containing_nat():
...
+
+
+def test_infer_dtype_pyarrow_dtype(data):
+    res = lib.infer_dtype(data)
+    assert res != "unknown-array"
+
+    if data._hasna and res in ["floating", "datetime64", "timedelta64"]:
+        # FIXME: in infer_dtype pd.NA is not ignored in these cases even
+        #  with skipna=True in the list(data) check below
+        return
```
Comment (mroeschke) [inline pandas/tests/extension/test_arrow.py:2887]: Would we turn this into an xfail so we know it needs addressing?

Comment (jbrockmendel) [reply pandas/tests/extension/test_arrow.py:2887]: good idea, will update

Comment (jbrockmendel) [reply pandas/tests/extension/test_arrow.py]: updated+green

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**